Corrected region not being used in call to EnableCDNOnContainer (Issu…

husterk opened the pull request:
fixes #515

sharwell commented:
Hi @husterk,

Thanks for taking the time to help correct these issues! :smile: 

It looks like you fell into the (common) sneaky trap of updating the source branch for this pull request after you created the pull request. When this happens, GitHub automatically updates the pull request to include the additional commits.

You might not need to make any changes to this pull request; we'll review it as a proposed solution for 3 issues (#515, #517, #519) and try to only ask you to break it up if one of them has a problem. However, keep in mind that if you update the branch again it could force us to restart the review. :wink:

sharwell commented on build/build.proj:
```diff
@@ -3,6 +3,7 @@
 
   <PropertyGroup>
     <Configuration>Debug</Configuration>
+    <Platform>&quot;Any CPU&quot;</Platform>
```
:exclamation: It looks like this flag was lost during the migration from **build.ps1** to **build.proj**. The [previous build script](https://github.com/openstacknetsdk/openstack.net/blob/8ad5dabe635942f9013705193c2f95331b1ce905/build/build.ps1#L112) used the **Mixed Platforms** platform:

```xml
<Platform>Mixed Platforms</Platform>
```

husterk replied:
I will switch the Platform tag to be Mixed Platforms instead of Any CPU. However, I am fairly certain that it would not work unless I had added quotes around the "Mixed Platforms". I will take a look at this again though.

sharwell commented on build/build.proj:
```diff
@@ -21,7 +23,7 @@
 
   <Target Name="Build" DependsOnTargets="RestorePackages">
-    <Exec Command="$(MSBuild) ..\src\openstack.net.sln /p:Configuration=$(Configuration) /nologo /v:minimal" WorkingDirectory="$(MSBuildThisFileDirectory)" />
+    <Exec Command="$(MSBuild) ..\src\openstack.net.sln /p:Configuration=$(Configuration) /p:Platform=$(Platform) /nologo /v:minimal" WorkingDirectory="$(MSBuildThisFileDirectory)" />
```
:memo: If you change the `<Platform>` property to remove the quotes as described previously, this line will need to quote the associated command line argument.

husterk replied:
Noted. Thanks.

sharwell commented on build/build.proj:
```diff
@@ -30,7 +32,7 @@
 
     <Exec Command="$(NuGet) restore ..\src\Documentation\Documentation.sln" WorkingDirectory="$(MSBuildThisFileDirectory)" />
-    <Exec Command="(set SHFBROOT=$(SHFBROOT)) &amp; $(MSBuild) ..\src\Documentation\Documentation.sln /p:Configuration=$(Configuration) /nologo /v:minimal" WorkingDirectory="$(MSBuildThisFileDirectory)" />
+    <Exec Command="(set SHFBROOT=$(SHFBROOT)) &amp; $(MSBuild) ..\src\Documentation\Documentation.sln /p:Configuration=$(Configuration) /p:Platform=$(Platform) /nologo /v:minimal" WorkingDirectory="$(MSBuildThisFileDirectory)" />
```
:memo: If you change the `<Platform>` property to remove the quotes as described previously, this line will need to quote the associated command line argument.

sharwell commented on build/build.proj:
```diff
@@ -12,6 +13,7 @@
   <Target Name="CI">
     <PropertyGroup>
       <Configuration>Release</Configuration>
+      <Platform>&quot;Any CPU&quot;</Platform>
```
:bulb: This line isn't necessary, because only one platform is ever used by the build and the definition added above will be applied in all cases. I believe duplicating the definition increases the risk that the property will become incorrect in the future.

husterk commented:
Thanks Sam. I didn't realize that the master branch behaved this way when I
forked a repo. I am somewhat new to GitHub. I figured this out after the
fact and I will be sure to issue pull requests against stand alone bug fix
branches from now on. Thanks for explaining why this occurred and let me
know if there is anything else that I can do to help.

Keith
On Jun 27, 2015 11:40 PM, "Sam Harwell" <notifications@github.com> wrote:

> Hi @husterk <https://github.com/husterk>,
>
> Thanks for taking the time to help correct these issues! [image: :smile:]
>
> It looks like you fell into the (common) sneaky trap of updating the
> source branch for this pull request after you created the pull request.
> When this happens, GitHub automatically updates the pull request to include
> the additional commits.
>
> You might not need to make any changes to this pull request; we'll review
> it as a proposed solution for 3 issues (#515
> <https://github.com/openstacknetsdk/openstack.net/issues/515>, #517
> <https://github.com/openstacknetsdk/openstack.net/issues/517>, #519
> <https://github.com/openstacknetsdk/openstack.net/issues/519>) and try to
> only ask you to break it up if one of them has a problem. However, keep in
> mind that if you update the branch again it could force us to restart the
> review. [image: :wink:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openstacknetsdk/openstack.net/pull/516#issuecomment-116189639>
> .
>

sharwell commented on src/corelib/Providers/Rackspace/CloudFilesProvider.cs:
```diff
@@ -459,7 +459,7 @@ public IEnumerable<ContainerCDN> ListCDNContainers(int? limit = null, string mar
         public Dictionary<string, string> EnableCDNOnContainer(string container, long timeToLive, string region = null, CloudIdentity identity = null)
         {
-            return EnableCDNOnContainer(container, timeToLive, false, identity: identity);
+            return EnableCDNOnContainer(container, timeToLive, false, region, identity);
```
:+1:

sharwell commented:
:question: Do you think there's a way to add a command line argument to **build.cmd** which covers your **buildAndPackage.cmd** case? If so, we could simplify the change with one of the following:

1. Update **build.cmd** to "always do the right thing"
2. Update **build.cmd** to support your use case as an option, and [possibly] also add **buildAndPackage.cmd**. In this case, **buildAndPackage.cmd** would be nothing more than a one-liner that calls **build.cmd** with that flag.

:thought_balloon: It's not clear to me from the names **build.cmd** and **buildAndPackage.cmd** what each does. On the other hand, I also don't have better name(s) to propose.

husterk commented:
Yeah. I can take a look at this in the morning.

Keith
On Jun 27, 2015 11:55 PM, "Sam Harwell" <notifications@github.com> wrote:

> [image: :question:] Do you think there's a way to add a command line
> argument to *build.cmd* which covers your *buildAndPackage.cmd* case? If
> so, we could simplify the change with one of the following:
>
>    1. Update *build.cmd* to "always do the right thing"
>    2. Update *build.cmd* to support your use case as an option, and
>    [possibly] also add *buildAndPackage.cmd*. In this case,
>    *buildAndPackage.cmd* would be nothing more than a one-liner that
>    calls *build.cmd* with that flag.
>
> [image: :thought_balloon:] It's not clear to me from the names *build.cmd*
> and *buildAndPackage.cmd* what each does. On the other hand, I also don't
> have better name(s) to propose.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openstacknetsdk/openstack.net/pull/516#issuecomment-116190543>
> .
>

husterk commented:
I have updated the build script to support the 4 necessary build types: buildDebug, buildRelease, buildDebugAndPackage, and buildReleaseAndPackage. I have also added a build type called buildReleaseAndPublish that is not documented in the build help so that you can still push the package to NuGet. I will update the pull request shortly.

husterk commented:
New changes are in per your requests. Let me know if you need any further updates. I tested all associatd build types (except the BuildAndPackageAndPublish) and all executed cleanly. We will need to update the main readme.md file to describe the new build.cmd command line args if these changes are accepted.

husterk commented on build/build.proj:
```diff
@@ -30,7 +82,7 @@
 
     <Exec Command="$(NuGet) restore ..\src\Documentation\Documentation.sln" WorkingDirectory="$(MSBuildThisFileDirectory)" />
-    <Exec Command="(set SHFBROOT=$(SHFBROOT)) &amp; $(MSBuild) ..\src\Documentation\Documentation.sln /p:Configuration=$(Configuration) /nologo /v:minimal" WorkingDirectory="$(MSBuildThisFileDirectory)" />
+    <Exec Command="(set SHFBROOT=$(SHFBROOT)) &amp; $(MSBuild) ..\src\Documentation\Documentation.sln /p:Configuration=$(Configuration) /p:Platform=&quot;$(DocumentationPlatform)&quot; /nologo /v:minimal" WorkingDirectory="$(MSBuildThisFileDirectory)" />
```
Please note that I had to add a separate platform configuration for the documentation project since it did not support the "Mixed Platforms" platform configuration.

sharwell commented:
:bulb: If [this line of **appveyor.yml**](https://github.com/openstacknetsdk/openstack.net/blob/master/appveyor.yml#L4) is updated, the build will probably be fixed. :smile:

carolynvs commented:
@husterk Thanks for the issues and PR! :smiley:  

I would prefer that we separate the #515  bug fix from the build script changes as I would like to review and think about further. Would you please update the branch for this PR so that it only contains the original commit 3284626b1e4c97a9df16340ac4b4557f6189b833? 

If you run into trouble, I can split it out for you, just let me know.

husterk commented:
Closing this pull request to separate out the changes for the #515 fix from the #517 and #519 issues.
